File: mambo-protocol/src/main/scala/org/mambo/protocol/messages/game/prism/PrismFightAttackedMessage.scala





















// Generated on 05/03/2013 10:34:58
// Scala version by Blackrush
package org.mambo.protocol.messages

import org.mambo.core.io._
import org.mambo.protocol.{Message => BaseMessage, _}
import org.mambo.protocol.types._
import org.mambo.protocol.enums._

trait PrismFightAttackedMessage extends MessageDeserializer {
  val messageId = 5894

  protected trait Message extends BaseMessage with Serializable {
    val messageId = PrismFightAttackedMessage.messageId

    val worldX: Short
    val worldY: Short
    val mapId: Int
    val subAreaId: Short
    val prismSide: Byte

    def serialize(buf: Buffer) {
      buf.writeShort(worldX)
      buf.writeShort(worldY)
      buf.writeInt(mapId)
      buf.writeShort(subAreaId)
      buf.writeByte(prismSide)
    }

  }

  protected def create(worldX: Short, worldY: Short, mapId: Int, subAreaId: Short, prismSide: Byte): Target = throw new IllegalAccessException

  def deserialize(buf: Buffer) = {
    val worldX = buf.readShort()
    val worldY = buf.readShort()
    val mapId = buf.readInt()
    val subAreaId = buf.readShort()
    val prismSide = buf.readByte()

    create(worldX, worldY, mapId, subAreaId, prismSide)
  }

}

object PrismFightAttackedMessage extends PrismFightAttackedMessage {
  case class MessageImpl(worldX: Short, worldY: Short, mapId: Int, subAreaId: Short, prismSide: Byte) extends Message
  type Target = MessageImpl

  override def create(worldX: Short, worldY: Short, mapId: Int, subAreaId: Short, prismSide: Byte) = MessageImpl(worldX, worldY, mapId, subAreaId, prismSide)

  def apply(worldX: Short, worldY: Short, mapId: Int, subAreaId: Short, prismSide: Byte) = create(worldX, worldY, mapId, subAreaId, prismSide)
  def unapply(msg: MessageImpl) = MessageImpl.unapply(msg)
}

File: mambo-protocol/src/main/java/org/mambo/protocol/messages/PrismFightAttackedMessage.java


// Generated on 05/08/2013 19:37:57
package org.mambo.protocol.messages;

import java.util.*;
import org.mambo.protocol.types.*;
import org.mambo.protocol.enums.*;
import org.mambo.protocol.*;
import org.mambo.core.io.*;

public class PrismFightAttackedMessage extends NetworkMessage {
    public static final int MESSAGE_ID = 5894;
    
    @Override
    public int getNetworkMessageId() {
        return MESSAGE_ID;
    }
    
    public short worldX;
    public short worldY;
    public int mapId;
    public short subAreaId;
    public byte prismSide;
    
    public PrismFightAttackedMessage() { }
    
    public PrismFightAttackedMessage(short worldX, short worldY, int mapId, short subAreaId, byte prismSide) {
        this.worldX = worldX;
        this.worldY = worldY;
        this.mapId = mapId;
        this.subAreaId = subAreaId;
        this.prismSide = prismSide;
    }
    
    @Override
    public void serialize(Buffer buf) {
        buf.writeShort(worldX);
        buf.writeShort(worldY);
        buf.writeInt(mapId);
        buf.writeShort(subAreaId);
        buf.writeByte(prismSide);
    }
    
    @Override
    public void deserialize(Buffer buf) {
        worldX = buf.readShort();
        if (worldX < -255 || worldX > 255)
            throw new RuntimeException("Forbidden value on worldX = " + worldX + ", it doesn't respect the following condition : worldX < -255 || worldX > 255");
        worldY = buf.readShort();
        if (worldY < -255 || worldY > 255)
            throw new RuntimeException("Forbidden value on worldY = " + worldY + ", it doesn't respect the following condition : worldY < -255 || worldY > 255");
        mapId = buf.readInt();
        subAreaId = buf.readShort();
        if (subAreaId < 0)
            throw new RuntimeException("Forbidden value on subAreaId = " + subAreaId + ", it doesn't respect the following condition : subAreaId < 0");
        prismSide = buf.readByte();
    }
    
}
